learntools/ml_intermediate/ex6.py

```diff
@@ -5,6 +5,33 @@
 from sklearn.model_selection import train_test_split
 
 from learntools.core import *
+
+# Load the data
+X = pd.read_csv('../input/train.csv', index_col='Id')
+X_test_full = pd.read_csv('../input/test.csv', index_col='Id')
+X.dropna(axis=0, subset=['SalePrice'], inplace=True)
+y = X.SalePrice              
+X.drop(['SalePrice'], axis=1, inplace=True)
+X_train_full, X_valid_full, y_train, y_valid = train_test_split(X, y, train_size=0.8, test_size=0.2,
+                                                                random_state=0)
+low_cardinality_cols = [cname for cname in X_train_full.columns if X_train_full[cname].nunique() < 10 and 
+                        X_train_full[cname].dtype == "object"]
+numeric_cols = [cname for cname in X_train_full.columns if X_train_full[cname].dtype in ['int64', 'float64']]
+my_cols = low_cardinality_cols + numeric_cols
+X_train = X_train_full[my_cols].copy()
+X_valid = X_valid_full[my_cols].copy()
+X_test = X_test_full[my_cols].copy()
+X_train = pd.get_dummies(X_train)
+X_valid = pd.get_dummies(X_valid)
+X_test = pd.get_dummies(X_test)
+X_train, X_valid = X_train.align(X_valid, join='left', axis=1)
+X_train, X_test = X_train.align(X_test, join='left', axis=1)
+
+# Define and train basic model
+basic_model = XGBRegressor(random_state=0)
+basic_model.fit(X_train, y_train)
```

Comment:
@alexisbcook Do you know how long this takes?  

If it takes more than a couple seconds time, we should think about ways to do go a good job checking without making the user wait.

Reply:
on my mac, just over two seconds
<img width="508" alt="Screen Shot 2019-04-01 at 1 09 19 PM" src="https://user-images.githubusercontent.com/10624937/55353031-90e4bc00-547f-11e9-80a1-7793308fe477.png">